.pre-commit-config.yaml

```diff
@@ -213,7 +213,9 @@ repos:
       - id: pyupgrade
         name: Upgrade Python code automatically
         args: ["--py37-plus"]
-        exclude: ^airflow/_vendor/
+        # Test Python tests if different kinds of typing including one that does not have
+        # __future__ annotations, so it shuld be left without automated upgrade
+        exclude: ^airflow/_vendor/|^test/decorators/test_python.py^
```

Comment:
Can you explain (here, not comment) why this decorator can't have the annotation?

Reply:
Because it checks in runtime the type of annotation used and explicitly wants to use Dict which otherwise would be removed by pyupgrade (pyupgrade is not very configurable and I could not find a way to exclude specific lines from pyupgrade checks - I only know how to exclude the whole file. The test in question is this:

```
    @pytest.mark.parametrize(
        "annotation",
        [
            "dict",
            pytest.param(
                "dict[str, int]",
                marks=pytest.mark.skipif(
                    sys.version_info < (3, 9),
                    reason="PEP 585 is implemented in Python 3.9",
                ),
            ),
            "Dict",
            "Dict[str, int]",
        ],
    )
    def test_infer_multiple_outputs_using_dict_typing(self, resolve, annotation):
```

I think I even tried to replace`Dict` by `dict` in this case and it did not work out-of-the-box, so for the sake of "as small and as safe change as possible" I decided to exclude the whole file from pyupgrade and add appropriate comments (I did it after Pyupgrade already made changes, just backed-out the Dict changes.

There is even a comment at the top mentioning it:

```
from typing import Dict  # noqa: F401  # This is used by annotation tests.
```